internal/shim-sev/src/hostmap.rs

```diff
@@ -0,0 +1,204 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Provides a mapping of the shim physical addresses
+//! to the hypervisor host virtual addresses, needed
+//! for proxying syscalls.
+
+use crate::addr::{HostVirtAddr, ShimPhysUnencryptedAddr};
+
+use crate::spin::RWLocked;
+use crate::SHIM_CAN_PRINT;
+use core::alloc::Layout;
+use core::mem::{align_of, size_of};
+use core::sync::atomic::Ordering;
+use linked_list_allocator::Heap;
+use lset::Line;
+use primordial::{Address, Page as Page4KiB};
+use spinning::{Lazy, RwLockWriteGuard};
+
+/// The global [`HostMap`](HostMap) RwLock
+pub static HOSTMAP: Lazy<RWLocked<HostMap>> =
+    Lazy::new(|| RWLocked::<HostMap>::new(HostMap::new()));
+
+struct HostMemListPageHeader {
+    next: Option<&'static mut HostMemListPage>,
+}
+
+#[derive(Clone, Copy)]
+struct HostMemEntry {
```

Comment:
Any comment about what this struct's data means?

Reply:
Added more comments in the hope it clarifies the logic.